### ISSUE TITLE
ipv6 unsupported for no-ip

### ISSUE THREAD

**Issue by hmijail** [ddclient/ddclient#633]
Just so that Google & co. pick it up and save people's time...

I spent way too long trying to make sense of the documentation and config format to finally find that the info sent to No-IP is only IPv4, even if it's an empty string while the IPv6 is available.

**Comment by EricV** [comment]
I agree that examples for IPV4 only, IPV6 only, and dualstack are badly missing. I spend also way to much time trying to figure out a correct config file and I'm not even sure the one I have works although I have no error in log.

**Comment by misyntropy** [comment]
The no-ip ddclient provider is broken — ipv6 works if you use the dyndns2 provider with the no-ip server.

**Comment by beadon** [comment]
Thanks for logging this, and apologies for the slow follow-up.

This was a valid bug in **ddclient 3.x** — the `noip` protocol only sent IPv4. It has been fixed in **ddclient 4.x**, which added native IPv6 and dual-stack support to the `noip` protocol.

Under the hood, ddclient sends both addresses as `myip=<ipv4>,<ipv6>` (or just `myip=<ipv6>` for IPv6-only) over an IPv4 connection to `dynupdate.no-ip.com`. That host has no AAAA record by design — the IPv6 address travels as a parameter value, not as the transport — so IPv6 updates work fine regardless.

---

### Sample configs (ddclient 4.x)

**IPv4-only:**
```
protocol=noip,                                    \
usev4=webv4,                                      \
login=user@example.com,                           \
password=noip-password                            \
myhost.no-ip.biz
```

**IPv6-only:**
```
protocol=noip,                                    \
usev6=webv6,                                      \
login=user@example.com,                           \
password=noip-password                            \
myhost.no-ip.biz
```

**Dual-stack (IPv4 + IPv6):**
```
protocol=noip,                                    \
usev4=webv4,                                      \
usev6=webv6,                                      \
login=user@example.com,                           \
password=noip-password                            \
myhost.no-ip.biz
```

Replace `webv4`/`webv6` with `ifv4`/`ifv6` (plus `if=eth0`) if you prefer to read the address directly from a network interface rather than using a web-based IP detection service.

---

The `dyndns2` workaround noted by @misyntropy still works too, but the native `noip` protocol handles both IP versions correctly in 4.x, so there is no longer a reason to reach for it.

Closing this as resolved in 4.x — please reopen if you hit a regression.
